openassessment/data.py

```diff
@@ -333,4 +337,135 @@ def _use_read_replica(self, queryset):
+            " -- scored_at: ", `assessment`.`scored_at`,
+            " -- type: ", `assessment`.`score_type`,
+            " -- scorer_id: ", `assessment`.`scorer_id`,
+            IF(
+                `assessment`.`feedback` != "",
+                CONCAT(" -- overall_feedback: ", `assessment`.`feedback`),
+                ""
+            )
+        )
+        SEPARATOR '\n'
+    )
+    FROM `assessment_assessment` AS `assessment`
+    WHERE `assessment`.`submission_uuid`=`sub`.`uuid`
+    ORDER BY `assessment`.`scored_at` ASC
+) AS `assessments`,
+(
+    SELECT GROUP_CONCAT(
+        CONCAT(
+            "Assessment #", `assessment`.`id`,
+            " -- ", `criterion`.`label`,
+            IFNULL(CONCAT(": ", `option`.`label`, " (", `option`.`points`, ")"), ""),
+            IF(
+                `assessment_part`.`feedback` != "",
+                CONCAT(" -- feedback: ", `assessment_part`.`feedback`),
+                ""
+            )
+        )
+        SEPARATOR '\n'
+    )
+    FROM `assessment_assessment` AS `assessment`
+    JOIN `assessment_assessmentpart` AS `assessment_part`
+    ON `assessment_part`.`assessment_id`=`assessment`.`id`
+    JOIN `assessment_criterion` AS `criterion`
+    ON `criterion`.`id`=`assessment_part`.`criterion_id`
+    LEFT JOIN `assessment_criterionoption` AS `option`
+    ON `option`.`id`=`assessment_part`.`option_id`
+    WHERE `assessment`.`submission_uuid`=`sub`.`uuid`
+    ORDER BY `assessment`.`scored_at` ASC, `criterion`.`order_num` DESC
+) AS `assessments_parts`,
+(
+    SELECT `created_at`
+    FROM `submissions_score` AS `score`
+    WHERE `score`.`submission_id`=`sub`.`id`
+    ORDER BY `score`.`created_at` DESC LIMIT 1
+) AS `final_score_given_at`,
+(
+    SELECT `points_earned`
+    FROM `submissions_score` AS `score`
+    WHERE `score`.`submission_id`=`sub`.`id`
+    ORDER BY `score`.`created_at` DESC LIMIT 1
+) AS `final_score_points_earned`,
+(
+    SELECT `points_possible`
+    FROM `submissions_score` AS `score`
+    WHERE `score`.`submission_id`=`sub`.`id`
+    ORDER BY `score`.`created_at` DESC LIMIT 1
+) AS `final_score_points_possible`,
+(
+    SELECT GROUP_CONCAT(`feedbackoption`.`text` SEPARATOR '\n')
+    FROM `assessment_assessmentfeedbackoption` AS `feedbackoption`
+    JOIN `assessment_assessmentfeedback_options` AS `feedback_join`
+    ON `feedback_join`.`assessmentfeedbackoption_id`=`feedbackoption`.`id`
+    JOIN `assessment_assessmentfeedback` AS `feedback`
+    ON `feedback`.`id`=`feedback_join`.`assessmentfeedback_id`
+    WHERE `feedback`.`submission_uuid`=`sub`.`uuid`
+) AS `feedback_options`,
+(
+    SELECT `feedback_text`
+    FROM `assessment_assessmentfeedback` as `feedback`
+    WHERE `feedback`.`submission_uuid`=`sub`.`uuid`
+    LIMIT 1
+) AS `feedback`
+FROM `submissions_submission` AS `sub`
+JOIN `submissions_studentitem` AS `student` ON `sub`.`student_item_id`=`student`.`id`
+WHERE `student`.`item_type`="openassessment" AND `student`.`course_id`=%s
```

Comment:
Do we need to filter out data with status deleted from the submissions api, or should that already be properly handled by relying on valid ORA data?  You may want to test Clear Student State with the data download to verify.